OpenRA.Mods.RA/Traits/Disguise.cs

```diff
@@ -61,18 +62,14 @@ class DisguiseInfo : ITraitInfo
 	{
 		[VoiceReference] public readonly string Voice = "Action";
 
+		[Desc("Upgrades to grant when disguised.")]
+		public readonly string[] Upgrades = { "disguise" };
```

Comment:
This needs a `[UpgradeGrantedReference]` now. See https://github.com/OpenRA/OpenRA/pull/8321/